src/genie/libs/parser/asa/show_access_list.py

```diff
@@ -0,0 +1,187 @@
+                continue
+            
+            m = p2.match(line)
+            if m:
+                groups = m.groupdict()
+                acl_name = groups['name']
+                line_num = groups['entry']
+                if not acl_name in ret_dict['access-list'].keys() :
+                    ret_dict['access-list'][acl_name] = {}
+                if not 'entry' in ret_dict['access-list'][acl_name].keys():
+                    ret_dict['access-list'][acl_name]['entry'] = {}
+                line_entry = {}
+                self.add_if_in_groups(line_entry, groups, ['action', 'protocol', 'hitcnt', 'acl_hash'])
+                line_entry['source'] = {}
+                line_entry['destination'] = {}
+
+                self.add_host(line_entry['source'], groups, 'src_')
+                self.add_host(line_entry['destination'], groups, 'dst_')
+
+                if 'hitcnt' in line_entry:
+                    line_entry['hitcnt'] = int(line_entry['hitcnt'])
+                ret_dict['access-list'][acl_name]['entry'][line_num] = line_entry
+                continue
+
+            m = p3.match(line)
+            if m:
+                groups = m.groupdict()
+                line_num = groups['entry']
+                if not acl_name in ret_dict['access-list'].keys() :
+                    ret_dict['access-list'][acl_name] = {}
+                if not 'entry' in ret_dict['access-list'][acl_name].keys():
+                    ret_dict['access-list'][acl_name]['entry'] = {}
+                ret_dict['access-list'][acl_name]['entry'][line_num] = {}
+                ret_dict['access-list'][acl_name]['entry'][line_num]['remark'] = groups['remark']
+                continue
+
+
+            m = p4.match(line)
+            if m:
+                groups = m.groupdict()
+                flows = {}
+                flows['total'] = int(groups['total'])
+                flows['denied'] = int(groups['denied'])
+                flows['deny_flow_max'] = int(groups['deny_flow_max'])
+                ret_dict['cached_acl_log_flows'] = flows
+                continue
+
+
+            m = p5.match(line)
+            if m:
+                groups = m.groupdict()
+                ret_dict['alert_interval'] = int(groups['alert_interval'])
+#            else:
+#                print('Not matching line: {}'.format(line))
+
+
+        if not(bool(ret_dict['access-list'])):
+            ret_dict = {}
+        return ret_dict
+
+    def add_if_in_groups(self, line_dict, groups, names):
+        for name in names:
+            if name in groups and not groups[name] is None:
+                line_dict[name] = groups[name]
+
+    def add_host(self, line_dict, groups, prefix):
+        names = ['host', 'any', 'object_group']
+        for name in names:
+            prefixed_name = prefix + name
+            if prefixed_name in groups and not groups[prefixed_name] is None:
+                line_dict[name] = groups[prefixed_name]
```

Comment:
Please do this logic within each regex match. There shouldnt be any more methods defined so that all parsers follow the same format